keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -333,36 +333,54 @@ or later. The dual-stack feature requires no changes to this API.
 The versions of CNI plugin binaries that must be used for proper dual-stack functionality (and IPv6 functionality in general) depend upon the version of the container runtime that is used in the cluster nodes (see [CNI issue #531](https://github.com/containernetworking/cni/issues/531) and [CNI plugins PR #113](https://github.com/containernetworking/plugins/pull/113)):
 
 ### Services
+Services depend on `--services-cluster-ip-range` for `ClusterIP` assignment. A dualstack cluster can have this flag configured as `--services-cluster-ip-rang=<cidr>,<cidr>` or `--services-cluster-ip-rang=<cidr>`. The following rules apply:
+
+1. The maximum allowed number of cidrs in this flag is two. Attempting to use more than two cidrs will generate failures during startup. 
+2. When configured with two cidrs then the values are expected to have different ip families. e.g `--services-cluster-ip-range=<ipv4-cidr>,<ipv6-cidr>` or `--services-cluster-ip-range=<ipv6-cidr>,<ipv4-cidr>`. Attempting to configure a cluster with two similar cidrs will generate failures during startup.
+3. A dualstack cluster can have a single cidr in this flag. That means the cluster will be to egress traffic on both ip families, but will *not* be ingress traffic on both ip families.
```

Comment:
No. it is correct.

The basic idea is, users may want to dualstack egress only. An example for that is a queue processor (connected to ipv6 and ipv4 exposed queues). Or users may want to have both egress and ingress. I can remove this entire statement.

Reply:
I think that this is not part of the design because clusters can do that right now without enabling dual stack, pods with dual stack addresses are going to use the corresponding IP family for their communication independently of what Kubernetes says.